setup.py

```diff
@@ -130,11 +134,9 @@ def get_extras_require() -> Dict[str, List[str]]:
             "pandas",
             "plotly>=4.0.0",
             "pytest",
-            "scikit-learn>=0.19.0,<0.23.0",
             "scikit-optimize",
             "xgboost",
             "keras",
```

Comment:
Don't we need to exclude Keras, tensorflow-datasets, and tensorboard for Python3.9?

Reply:
You are right. Fixed this in the latest commit

Reply:
> Could you use environment markers for scikit-learn and TensorFlow? ref: https://pip.pypa.io/en/stable/reference/pip_install/?highlight=python_version#requirement-specifiers
>

Made the changes in the latest commit.

> Also, I'm not sure but this pull request might conflict with #2528.

#2528 doesn't conflict with this PR .